dockstore-integration-testing/src/test/java/io/dockstore/webservice/WebhookIT.java

```diff
@@ -900,15 +902,44 @@ public void testTools() throws Exception {
         testingPostgres.runUpdateStatement("update workflow set ispublished = 't' where id = " + workflow.getId());
 
         Ga4Ghv20Api ga4Ghv20Api = new Ga4Ghv20Api(openApiClient);
-        final List<Tool> tools = ga4Ghv20Api.toolsGet(null, null, null, null, null, null, null, null, null, null, null, null, null);
+        final List<io.dockstore.openapi.client.model.Tool> tools = ga4Ghv20Api.toolsGet(null, null, null, null, null, null, null, null, null, null, null, null, null);
         assertEquals(2, tools.size());
 
-        final Tool tool = ga4Ghv20Api.toolsIdGet("github.com/DockstoreTestUser2/test-workflows-and-tools");
+        final io.dockstore.openapi.client.model.Tool tool = ga4Ghv20Api.toolsIdGet("github.com/DockstoreTestUser2/test-workflows-and-tools");
         assertNotNull(tool);
         assertEquals("CommandLineTool", tool.getToolclass().getDescription());
 
         final Tool trsWorkflow = ga4Ghv20Api.toolsIdGet(ToolsImplCommon.WORKFLOW_PREFIX + "/github.com/DockstoreTestUser2/test-workflows-and-tools");
         assertNotNull(trsWorkflow);
         assertEquals("Workflow", trsWorkflow.getToolclass().getDescription());
     }
+
+    @Test
+    public void testDuplicatePathsAcrossTables() throws Exception {
+        CommonTestUtilities.cleanStatePrivate2(SUPPORT, false);
+        final ApiClient webClient = getWebClient(BasicIT.USER_2_USERNAME, testingPostgres);
+        final io.dockstore.openapi.client.ApiClient openApiClient = getOpenAPIWebClient(BasicIT.USER_2_USERNAME, testingPostgres);
+        WorkflowsApi client = new WorkflowsApi(webClient);
+
+        try {
+            client.handleGitHubRelease(toolAndWorkflowRepo, BasicIT.USER_2_USERNAME, "refs/heads/duplicate-paths", installationId);
+            fail("Should not be able to create a workflow and apptool with the same path.");
```

Comment:
Would probably work, but could use some tests locking down behaviour with workflow/apptool duplicating a service.

Reply:
added a line in the test for when there's a service in the yml too to verify you can create one without adding the name field.